landing-page/content/common/spark-quickstart.md

```diff
@@ -0,0 +1,370 @@
+df.writeTo("prod.nyc.taxis").append()
+```
+{{% /tabcontent %}}
+{{% tabcontent "PySpark" %}}
+```py
+schema = spark.table("prod.nyc.taxis").schema
+data = [
+    (1, 1000371, 1.8, 15.32, "N"),
+    (2, 1000372, 2.5, 22.15, "N"),
+    (2, 1000373, 0.9, 9.01, "N"),
+    (1, 1000374, 8.4, 42.13, "Y")
+  ]
+df = spark.createDataFrame(data, schema)
+df.writeTo("prod.nyc.taxis").append()
+```
+{{% /tabcontent %}}
+{{% /codetabs %}}
+
+### Reading
+
+To read a table, simply use the Iceberg table's name:
+
+{{% codetabs "SelectData" %}}
+{{% addtab "SparkSQL" checked %}}
+{{% addtab "SparkShell" %}}
+{{% addtab "PySpark" %}}
+{{% tabcontent "SparkSQL"  %}}
+```sql
+SELECT * FROM prod.nyc.taxis;
+```
+{{% /tabcontent %}}
+{{% tabcontent "SparkShell" %}}
+```scala
+val df = spark.table("prod.nyc.taxis").show()
+```
+{{% /tabcontent %}}
+{{% tabcontent "PySpark" %}}
+```py
+df = spark.table("prod.nyc.taxis").show()
+```
+{{% /tabcontent %}}
+{{% /codetabs %}}
+
+For [inspecting tables](../spark-queries#inspecting-tables) SQL is recommended. To view all of the snapshots in a table, use the `snapshots` metadata table:
+```sql
+SELECT committed_at, snapshot_id, parent_id, operation FROM prod.nyc.taxis.snapshots;
+```
+```
+2022-04-25 23:23:49.692	6317817083814616920	NULL	append
+```
+
+### Row-Level SQL Updates
+
+Iceberg also adds row-level SQL updates to Spark, [`MERGE INTO`](../spark-writes#merge-into) and [`DELETE FROM`](../spark-writes#delete-from). For example,
+if you wanted to merge in records from a table named `prod.nyc.updates`, you could run a `MERGE INTO` command to merge the updates into the `prod.nyc.taxis`
+table and simply add the `Fare_amount` for records where the `VendorID` and `Trip_ID` already exist.
+
+First, create the `updates` table.
+```sql
+CREATE TABLE prod.nyc.updates
+(
+  VendorID bigint,
+  TripID bigint,
+  Trip_distance float,
+  Fare_amount double,
+  Store_and_fwd_flag string
+)
+USING iceberg;
+```
+Next, insert a few records.
+```sql
+INSERT INTO prod.nyc.updates
+VALUES (1, 1000371, 1.8, 0.50, 'N'), (3, 1000375, 8.9, 37.15, 'Y');
+```
+Finally, merge the `updates` table into the `taxis` table.
+```sql
+MERGE INTO prod.nyc.taxis t USING (SELECT * FROM prod.nyc.updates) u ON t.VendorID = u.VendorID and t.TripID = u.TripID
+WHEN MATCHED THEN UPDATE SET t.Fare_amount = t.Fare_amount + u.Fare_amount
+WHEN NOT MATCHED THEN INSERT *;
+```
+Querying the `taxis` table, you can see that the fare amount of `0.50` has been added to `15.32` for `VendorID = 1 AND TripID = 1000371`.
+```sql
+SELECT VendorID, TripID, Fare_amount
+FROM prod.nyc.taxis
+WHERE VendorID = 1 AND TripID = 1000371;
+```
+```
+1	1000371	15.82
+```
+{{< hint warning >}}
+Iceberg supports writing DataFrames using the new [v2 DataFrame write API](../spark-writes#writing-with-dataframes). The old `write` API is supported, but _not_ recommended.
+{{< /hint >}}
+
+## Adding Iceberg to your Existing Spark Environment
```

Comment:
Are the steps related to adding iceberg extension to the spark environment & setup of the catalog not supposed to be done before creating data & insterting data into the table?

Reply:
Yes and this is a good point. Since the docker environment already sets up the catalog, it feels better to jump right into using Iceberg. Some feedback I've received is that when someone new comes to the site and wants to spend 10-20 minutes just playing around with Iceberg, they're thrown into a ton of configuration/setup options that are cognitively taxing when you just want to try it out. (What's a catalog? Which kind of catalog should I choose? Where should my warehouse path be? Should this be a local or S3 path?, etc)

I thought it would still be good to include this at the end for the subset of Iceberg newbies who are already very familiar with Spark and are maybe coming here with a spark cluster already setup to which they want to just add in Iceberg. What do you think?

Reply:
I thought as well that this ordering was done intentionally in order to get the newbies user onboard as soon as possible in the document.

Can you then add an introduction about the class notation?

`prod.nyc.taxis` can be hard to swallow the first time.

what is `prod` ? spark catalog name
what is `nyc` ? schema name

Reply:
You're totally right, I added in:

> To create your first Iceberg table in Spark, run a `CREATE TABLE` command. In the following example, we'll create a table using `prod.nyc.taxis` where `prod` is the catalog name, `nyc` is the schema name, and `taxis` is the table name.